modules/twinkleblock.js

```diff
@@ -1692,12 +1558,17 @@ Twinkle.block.callback.toggle_ds_reason = function twinkleblockCallbackToggleDSR
 		new RegExp(' ?\\(\\[\\[' + Twinkle.block.dsReason + '\\]\\]\\)'), ''
 	);
 
-	Twinkle.block.dsReason = Twinkle.block.dsinfo[this.options[this.selectedIndex].label].page;
-	if (!this.value) {
-		this.form.reason.value = reason;
-	} else {
-		this.form.reason.value = reason + ' ([[' + Twinkle.block.dsReason + ']])';
-	}
+	// grab discretionary sanctions list from en-wiki
+	Morebits.wiki.getCachedJson('Template:Ds/topics.json').then(function(dsinfo) {
```

Comment:
This is probably a nitpick as it's anyway cached, ... but still fetching it a second time seems bad. 

I'd suggest using this approach instead:

Twinkle.block.dsinfo = Morebits.wiki.getCachedJson('Template:Ds/topics.json'); // this is a promise

// 1st use 
Twinkle.block.dsinfo.then(...)

// 2nd use 
Twinkle.block.dsinfo.then(...)